source/TextBlade.Core/Battle/TurnBasedBattleSystem.cs

```diff
@@ -104,9 +104,9 @@ private void PopulateMonsters(List<string> monsterNames)
                 {
                     var skillName = token.Value<string>("Name");
                     var probability = token.Value<double>("Probability");
-                    var skill = Skill.GetSkill(skillName.ToString());
+                    var skill = Skill.GetSkill(skillName??"");
```

Comment:
This check doesn't make sense. This value should never be null.

I'm not sure if it's already done; please make sure GetSkill throws if the skill isn't found.

Reply:
The value can be null: token.Value<string>("Name") returns default (i.e. null) if the key isn't found.

Reply:
In the context of this code, we've already validated the skill data elsewhere (on load: see static constructor for ... I think the `Skill` class) and guaranteed it's not null.

I mean, you could technically say the same ting about all the other properties here. If you think it's safer/clearer to use `ArgumentNullException.ThrowIfNull(...)` for the bunch, that's cool with me.